docs/DeveloperGuide.md

```diff
@@ -382,33 +369,40 @@ The timings that the persons are free at the specified date will be displayed.
 
-The parsing of findCommonTiming command is handled by the following classes:
-* `AddressBookParser`
-    * Checks that the user input contains the FindCommonTimingCommand.COMMAND_WORD and calls `FindCommonTimingParser#parse()`.
-* `FindCommonTimingParser`
-    * Parses the user input to extract the required arguments.
-    * Creates a new `IsTagInPersonPredicate` object that will help check if persons in the address book have the tag that the user has inputted.
-    * Returns a `FindCommonTimingCommand` to be executed by the `LogicManager`.
+The `parse` method of `FindCommonTimingCommandParser` receives the user input and extracts the required arguments. It then creates a predicate object that will help check if the contact has the user-inputted tag attached to the contact. The user-inputted date will be used to retrieve events occuring on the same day such that the free timings can be determined. 
+
+A successful execuction of the `findCommonTiming` command is described as follows:
 
-Given below is an example usage scenario and explanation on how the `findCommonTiming` command behaves at each step.
+1. `FindCommonTimingCommand` uses the predicate prepared during parsing to filter the list of persons in `Model`.
+2. Events occuring on a certain day are then retrieved using the `getEventsAtDate` function.
+3. The day will be represented as an array of 48 time slots, with each element of the array representing a 30-minute timeslot.
+4. TheThe `blockTimeSlots` function is used to ensure that respective timeslots are set as busy according to when events occur.
+5. Free time slots will be appended to a string that is due to be returned in the `CommandResult`.
+6. A `CommandResult` with the timeslots that the persons are free will be returned (timeslots are in intervals of 30 minutes). 
+These timeslots will then be displayed to the user.
 
-1. The user enters `findCommonTiming t/friends da/2022-03-04` to find the common timings that the persons who share the same tag are free. 
-The arguments `t/friends da/2022-03-04` are passed to the `findCommonTimingParser` through its `parse` method call.
+<img src="images/FindCommonTimingSequenceDiagram.png" />
 
-2. The user input `t/friends da/2022-03-04` will be checked to ensure that empty input is not given.
+### Design Considerations
 
-3. A new `IsTagInPersonPredicate` object is created and encapsulated by a new `FindCommonTiming` object.
+**Aspect: How FindCommonTiming executes**
+* **Alternative 1(current implementation)**: Retrieve all events' starting and ending times. Use an integer array to represent 24 hours in a day, and change the value of array elements at specific indices of the array according to event start time and duration of event. After all events are processed, print common free timings according to which elements of the array indicate a certain time slot to be free.
```

Comment:
"Partition the day into time blocks. For every event that belongs to a person with the specified tag, get the start and end times of the event and block the relevant time blocks between the 2 timings to indicate that at least 1 person is unavailable during the time blocks. At the end, return the time blocks that have not been blocked."